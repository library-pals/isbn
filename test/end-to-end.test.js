import Isbn from "../src/index.js";

describe("End to end", () => {
  let isbn;
  beforeEach(() => {
    isbn = new Isbn();
  });

  afterEach(() => {
    isbn = null;
  });

  it("default", async () => {
    await expect(isbn.resolve("9780374104092")).resolves.toMatchInlineSnapshot(`
      {
        "authors": [
          "Jeff VanderMeer",
        ],
        "categories": [
          "Fiction",
        ],
        "description": "Describes the 12th expedition to “Area X,” a region cut off from the continent for decades, by a group of intrepid women scientists who try to ignore the high mortality rates of those on the previous 11 missions. Original. 75,000 first printing.",
        "isbn": "9780374104092",
        "link": "https://books.google.com/books/about/Annihilation.html?hl=&id=2cl7AgAAQBAJ",
        "pageCount": 209,
        "printType": "BOOK",
        "thumbnail": "http://books.google.com/books/content?id=2cl7AgAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
        "title": "Annihilation",
      }
    `);
  }, 10_000);

  it("openlibrary", async () => {
    isbn.provider([isbn.PROVIDER_NAMES.OPENLIBRARY]);
    await expect(isbn.resolve("9780374104092")).resolves.toMatchInlineSnapshot(`
      {
        "authors": [
          "Jeff VanderMeer",
        ],
        "categories": [
          "Nebula Award Winner",
          "award:nebula_award=novel",
          "award:nebula_award=2015",
          "Discoveries in geography",
          "Fiction",
          "Scientists",
          "Science-Fiction",
          "Suspense fiction",
          "Paranormal fiction",
          "Women scientists",
          "Science fiction",
          "Fantasy fiction",
          "Mystery fiction",
          "Adventure fiction",
          "Exploration",
          "Amerikanisches Englisch",
          "Fiction, science fiction, action & adventure",
          "Fiction, suspense",
          "Action & Adventure",
          "Dystopian",
          "Fantasy",
          "Extrasensory perception",
          "Literary",
          "Suspense",
          "Thrillers",
          "General",
          "Pollution",
          "horror",
          "body horror",
          "alien invasion",
          "nyt:trade-fiction-paperback=2018-03-18",
          "New York Times bestseller",
          "Explorers",
          "Secrecy",
          "Scientific expeditions",
          "Psychic ability",
          "Fiction, thrillers, suspense",
          "Discoveries in geography--fiction",
          "Scientists--fiction",
          "Women scientists--fiction",
          "Ps3572.a4284 a84 2014",
          "813/.54",
        ],
        "description": "Area X has been cut off from the rest of the continent for decades. Nature has reclaimed the last vestiges of human civilization. The twelfth expedition arrives expecting the unexpected, and Area X delivers. They discover a massive topographic anomaly and life-forms that surpass understanding. But it's the surprises that came across the border with them, and the secrets the expedition members are keeping from one another that change everything.",
        "isbn": "9780374104092",
        "link": "https://openlibrary.org/books/OL31444108M",
        "pageCount": 208,
        "printType": "BOOK",
        "thumbnail": "https://covers.openlibrary.org/b/id/10520611-L.jpg",
        "title": "Annihilation",
      }
    `);
  }, 10_000);

  it("google", async () => {
    isbn.provider([isbn.PROVIDER_NAMES.GOOGLE]);
    await expect(isbn.resolve("9780374104092")).resolves.toMatchInlineSnapshot(`
      {
        "authors": [
          "Jeff VanderMeer",
        ],
        "categories": [
          "Fiction",
        ],
        "description": "Describes the 12th expedition to “Area X,” a region cut off from the continent for decades, by a group of intrepid women scientists who try to ignore the high mortality rates of those on the previous 11 missions. Original. 75,000 first printing.",
        "isbn": "9780374104092",
        "link": "https://books.google.com/books/about/Annihilation.html?hl=&id=2cl7AgAAQBAJ",
        "pageCount": 209,
        "printType": "BOOK",
        "thumbnail": "http://books.google.com/books/content?id=2cl7AgAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
        "title": "Annihilation",
      }
    `);
  }, 10_000);
});
