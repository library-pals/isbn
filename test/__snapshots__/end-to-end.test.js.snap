// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End to end Annihilation { name: 'default', providers: [] } 1`] = `
{
  "authors": [
    "Jeff VanderMeer",
  ],
  "bookProvider": "Google Books",
  "categories": [
    "Fiction",
    "Fiction / General",
    "Fiction / Fantasy / General",
    "Fiction / Horror",
    "Fiction / Literary",
    "Fiction / Science Fiction / General",
    "Fiction / Science Fiction / Action & Adventure",
    "Fiction / Thrillers / Suspense",
    "Fiction / Dystopian",
  ],
  "description": "Area X has claimed the lives of members of eleven expeditions. The twelfth expedition consisting of four women hopes to map the terrain and collect specimens; to record all their observations, scientific and otherwise, of their surroundings and of one another; and, above all, to avoid being contaminated by Area X itself.",
  "format": "book",
  "isbn": "9780374104092",
  "language": "en",
  "link": "https://books.google.com/books/about/Annihilation.html?hl=&id=2cl7AgAAQBAJ",
  "pageCount": 209,
  "publishedDate": "2014-02-04",
  "publisher": "Macmillan",
  "thumbnail": "http://books.google.com/books/publisher/content?id=2cl7AgAAQBAJ&printsec=frontcover&img=1&zoom=6&edge=curl&source=gbs_api",
  "title": "Annihilation",
}
`;

exports[`End to end Annihilation { name: 'google', providers: [Array] } 1`] = `
{
  "authors": [
    "Jeff VanderMeer",
  ],
  "bookProvider": "Google Books",
  "categories": [
    "Fiction",
    "Fiction / General",
    "Fiction / Fantasy / General",
    "Fiction / Horror",
    "Fiction / Literary",
    "Fiction / Science Fiction / General",
    "Fiction / Science Fiction / Action & Adventure",
    "Fiction / Thrillers / Suspense",
    "Fiction / Dystopian",
  ],
  "description": "Area X has claimed the lives of members of eleven expeditions. The twelfth expedition consisting of four women hopes to map the terrain and collect specimens; to record all their observations, scientific and otherwise, of their surroundings and of one another; and, above all, to avoid being contaminated by Area X itself.",
  "format": "book",
  "isbn": "9780374104092",
  "language": "en",
  "link": "https://books.google.com/books/about/Annihilation.html?hl=&id=2cl7AgAAQBAJ",
  "pageCount": 209,
  "publishedDate": "2014-02-04",
  "publisher": "Macmillan",
  "thumbnail": "http://books.google.com/books/publisher/content?id=2cl7AgAAQBAJ&printsec=frontcover&img=1&zoom=6&edge=curl&source=gbs_api",
  "title": "Annihilation",
}
`;

exports[`End to end Annihilation { name: 'openlibrary', providers: [Array] } 1`] = `
{
  "authors": [
    "Jeff VanderMeer",
  ],
  "bookProvider": "Open Library",
  "categories": [
    "Nebula Award Winner",
    "award:nebula_award=novel",
    "award:nebula_award=2015",
    "Discoveries in geography",
    "Fiction",
    "Scientists",
    "Science-Fiction",
    "Suspense fiction",
    "Paranormal fiction",
    "Women scientists",
    "Science fiction",
    "Fantasy fiction",
    "Mystery fiction",
    "Adventure fiction",
    "Exploration",
    "Amerikanisches Englisch",
    "Fiction, science fiction, action & adventure",
    "Fiction, suspense",
    "Action & Adventure",
    "Dystopian",
    "Fantasy",
    "Extrasensory perception",
    "Literary",
    "Suspense",
    "Thrillers",
    "General",
    "Pollution",
    "horror",
    "body horror",
    "alien invasion",
    "nyt:trade-fiction-paperback=2018-03-18",
    "New York Times bestseller",
    "Explorers",
    "Secrecy",
    "Scientific expeditions",
    "Psychic ability",
    "Fiction, thrillers, suspense",
    "Discoveries in geography--fiction",
    "Scientists--fiction",
    "Women scientists--fiction",
    "Ps3572.a4284 a84 2014",
    "813/.54",
  ],
  "description": "Area X has been cut off from the rest of the continent for decades. Nature has reclaimed the last vestiges of human civilization. The twelfth expedition arrives expecting the unexpected, and Area X delivers. They discover a massive topographic anomaly and life-forms that surpass understanding. But it's the surprises that came across the border with them, and the secrets the expedition members are keeping from one another that change everything.",
  "format": "book",
  "isbn": "9780374104092",
  "language": "en",
  "link": "https://openlibrary.org/books/OL31444108M",
  "pageCount": 208,
  "publishedDate": "2014",
  "publisher": "Farrar, Straus and Giroux",
  "thumbnail": "https://covers.openlibrary.org/b/id/10520611-L.jpg",
  "title": "Annihilation",
}
`;

exports[`End to end Fantastic Mr. Fox { name: 'default', providers: [] } 1`] = `
{
  "authors": [
    "Roald Dahl",
  ],
  "bookProvider": "Google Books",
  "categories": [
    "Juvenile Fiction",
    "Juvenile Fiction / Animals / Foxes",
    "Juvenile Fiction / Fantasy & Magic",
  ],
  "description": "In this book you will find: Boggis an enormously fat man, a chicken farmer and a mean man. Bunce, a pot bellied dwarf, a duck-and-goose farmer and a nasty man. Bean, a thin man, a turkey-and-apple farmer and a beastly man. Badger, the most respectable and well-behaved animal in the district. Rat, a rude creature and a drunkard, and also a Mrs. Fox and her four children.",
  "format": "book",
  "isbn": "9780140328721",
  "language": "en",
  "link": "https://books.google.com/books/about/Fantastic_Mr_Fox.html?hl=&id=DlQbmJc5WlQC",
  "pageCount": 100,
  "publishedDate": "1988",
  "publisher": "Puffin",
  "thumbnail": "http://books.google.com/books/content?id=DlQbmJc5WlQC&printsec=frontcover&img=1&zoom=6&source=gbs_api",
  "title": "Fantastic Mr. Fox",
}
`;

exports[`End to end Fantastic Mr. Fox { name: 'google', providers: [Array] } 1`] = `
{
  "authors": [
    "Roald Dahl",
  ],
  "bookProvider": "Google Books",
  "categories": [
    "Juvenile Fiction",
    "Juvenile Fiction / Animals / Foxes",
    "Juvenile Fiction / Fantasy & Magic",
  ],
  "description": "In this book you will find: Boggis an enormously fat man, a chicken farmer and a mean man. Bunce, a pot bellied dwarf, a duck-and-goose farmer and a nasty man. Bean, a thin man, a turkey-and-apple farmer and a beastly man. Badger, the most respectable and well-behaved animal in the district. Rat, a rude creature and a drunkard, and also a Mrs. Fox and her four children.",
  "format": "book",
  "isbn": "9780140328721",
  "language": "en",
  "link": "https://books.google.com/books/about/Fantastic_Mr_Fox.html?hl=&id=DlQbmJc5WlQC",
  "pageCount": 100,
  "publishedDate": "1988",
  "publisher": "Puffin",
  "thumbnail": "http://books.google.com/books/content?id=DlQbmJc5WlQC&printsec=frontcover&img=1&zoom=6&source=gbs_api",
  "title": "Fantastic Mr. Fox",
}
`;

exports[`End to end Fantastic Mr. Fox { name: 'openlibrary', providers: [Array] } 1`] = `
{
  "authors": [
    "Roald Dahl",
  ],
  "bookProvider": "Open Library",
  "categories": [
    "Animals",
    "Hunger",
    "Open Library Staff Picks",
    "Juvenile fiction",
    "Children's stories, English",
    "Foxes",
    "Fiction",
    "Zorros",
    "Ficción juvenil",
    "Tunnels",
    "Interviews",
    "Farmers",
    "Children's stories",
    "Rats",
    "Welsh Authors",
    "English Authors",
    "Thieves",
    "Tricksters",
    "Badgers",
    "Children's fiction",
    "Foxes, fiction",
    "Underground",
    "Renards",
    "Romans, nouvelles, etc. pour la jeunesse",
    "Children's literature",
    "Plays",
    "Children's plays",
    "Children's stories, Welsh",
    "Agriculteurs",
    "Fantasy fiction",
    "Children's plays, English",
  ],
  "description": "The main character of Fantastic Mr. Fox is an extremely clever anthropomorphized fox named Mr. Fox. He lives with his wife and four little foxes. In order to feed his family, he steals food from the cruel, brutish farmers named Boggis, Bunce, and Bean every night.

Finally tired of being constantly outwitted by Mr. Fox, the farmers attempt to capture and kill him. The foxes escape in time by burrowing deep into the ground. The farmers decide to wait outside the hole for the foxes to emerge. Unable to leave the hole and steal food, Mr. Fox and his family begin to starve. Mr. Fox devises a plan to steal food from the farmers by tunneling into the ground and borrowing into the farmer's houses.

Aided by a friendly Badger, the animals bring the stolen food back and Mrs. Fox prepares a great celebratory banquet attended by the other starving animals and their families. Mr. Fox invites all the animals to live with him underground and says that he will provide food for them daily thanks to his underground passages. All the animals live happily and safely, while the farmers remain waiting outside in vain for Mr. Fox to show up.",
  "format": "book",
  "isbn": "9780140328721",
  "language": "en",
  "link": "https://openlibrary.org/books/OL7353617M",
  "pageCount": 96,
  "publishedDate": "October 1, 1988",
  "publisher": "Puffin",
  "thumbnail": "https://covers.openlibrary.org/b/id/8739161-L.jpg",
  "title": "Fantastic Mr. Fox",
}
`;

exports[`End to end The Ministry of Time (audiobook) { name: 'default', providers: [Array] } 1`] = `
{
  "authors": [
    "Kaliane Bradley",
  ],
  "bookProvider": "Libro.fm",
  "categories": [
    "Fiction",
    "Romance",
    "Science Fiction",
    "Fiction - Literary",
  ],
  "description": "In the near future, a civil servant is offered the salary of her dreams and is, shortly afterward, told what project she’ll be working on. A recently established government ministry is gathering “expats” from across history to establish whether time travel is feasible—for the body, but also for the fabric of space-time. She is tasked with working as a “bridge”: living with, assisting, and monitoring the expat known as “1847” or Commander Graham Gore. As far as history is concerned, Commander Gore died on Sir John Franklin’s doomed 1845 expedition to the Arctic, so he’s a little disoriented to be living with an unmarried woman who regularly shows her calves, surrounded by outlandish concepts such as “washing machines,” “Spotify,” and “the collapse of the British Empire.” But with an appetite for discovery, a seven-a-day cigarette habit, and the support of a charming and chaotic cast of fellow expats, he soon adjusts. Over the next year, what the bridge initially thought would be, at best, a horrifically uncomfortable roommate dynamic, evolves into something much deeper. By the time the true shape of the Ministry’s project comes to light, the bridge has fallen haphazardly, fervently in love, with consequences she never could have imagined. Forced to confront the choices that brought them together, the bridge must finally reckon with how—and whether she believes—what she does next can change the future. An exquisitely original and feverishly fun fusion of genres and ideas, The Ministry of Time asks: What does it mean to defy history, when history is living in your house? Kaliane Bradley’s answer is a blazing, unforgettable testament to what we owe each other in a changing world.",
  "format": "audiobook",
  "isbn": "9781797176888",
  "language": "en",
  "link": "https://libro.fm/audiobooks/9781797176888",
  "publishedDate": "2024-05-07",
  "publisher": "Simon & Schuster Audio",
  "thumbnail": "https://covers.libro.fm/9781797176888_1120.jpg",
  "title": "The Ministry of Time",
}
`;

exports[`End to end Transcendent Kingdom { name: 'default', providers: [] } 1`] = `
{
  "authors": [
    "Yaa Gyasi",
  ],
  "bookProvider": "Google Books",
  "categories": [
    "Fiction",
    "Fiction / Literary",
    "Fiction / Family Life / General",
    "Fiction / Family Life / Siblings",
    "Fiction / African American & Black / General",
    "Fiction / Cultural Heritage",
  ],
  "description": "Yaa Gyasi's stunning follow-up to her acclaimed national bestseller Homegoing is a powerful, raw, intimate, deeply layered novel about a Ghanaian family in Alabama. Gifty is a fifth year candidate in neuroscience at Stanford University's School of Medicine studying reward-seeking behaviour in mice and the neural circuits of depression and addiction. Her brother, Nana, was a gifted high school athlete who died of a heroin overdose after a knee injury left him hooked on OxyContin. Her suicidal mother is living in her bed. Gifty is determined to discover the scientific basis for the suffering she sees all around her. But even as she turns to hard science to unlock the mystery of her family's loss, she finds herself hungering for her childhood faith, and grappling with the evangelical church in which she was raised, whose promise of salvation remains as tantalizing as it is elusive. Transcendent Kingdom is a deeply moving portrait of a family of Ghanaian immigrants ravaged by depression and addiction and grief--a novel about faith, science, religion, love. Exquisitely written and emotionally searing, this is an exceptionally powerful follow-up to Gyasi's phenomenal debut.",
  "format": "book",
  "isbn": "9780385695176",
  "language": "en",
  "link": "https://books.google.com/books/about/Transcendent_Kingdom.html?hl=&id=gr4pywEACAAJ",
  "pageCount": 288,
  "publishedDate": "2020-09-15",
  "publisher": "Bond Street Books",
  "thumbnail": "http://books.google.com/books/content?id=gr4pywEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
  "title": "Transcendent Kingdom",
}
`;

exports[`End to end Transcendent Kingdom { name: 'google', providers: [Array] } 1`] = `
{
  "authors": [
    "Yaa Gyasi",
  ],
  "bookProvider": "Google Books",
  "categories": [
    "Fiction",
    "Fiction / Literary",
    "Fiction / Family Life / General",
    "Fiction / Family Life / Siblings",
    "Fiction / African American & Black / General",
    "Fiction / Cultural Heritage",
  ],
  "description": "Yaa Gyasi's stunning follow-up to her acclaimed national bestseller Homegoing is a powerful, raw, intimate, deeply layered novel about a Ghanaian family in Alabama. Gifty is a fifth year candidate in neuroscience at Stanford University's School of Medicine studying reward-seeking behaviour in mice and the neural circuits of depression and addiction. Her brother, Nana, was a gifted high school athlete who died of a heroin overdose after a knee injury left him hooked on OxyContin. Her suicidal mother is living in her bed. Gifty is determined to discover the scientific basis for the suffering she sees all around her. But even as she turns to hard science to unlock the mystery of her family's loss, she finds herself hungering for her childhood faith, and grappling with the evangelical church in which she was raised, whose promise of salvation remains as tantalizing as it is elusive. Transcendent Kingdom is a deeply moving portrait of a family of Ghanaian immigrants ravaged by depression and addiction and grief--a novel about faith, science, religion, love. Exquisitely written and emotionally searing, this is an exceptionally powerful follow-up to Gyasi's phenomenal debut.",
  "format": "book",
  "isbn": "9780385695176",
  "language": "en",
  "link": "https://books.google.com/books/about/Transcendent_Kingdom.html?hl=&id=gr4pywEACAAJ",
  "pageCount": 288,
  "publishedDate": "2020-09-15",
  "publisher": "Bond Street Books",
  "thumbnail": "http://books.google.com/books/content?id=gr4pywEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
  "title": "Transcendent Kingdom",
}
`;

exports[`End to end Transcendent Kingdom { name: 'openlibrary', providers: [Array] } 1`] = `
{
  "authors": [
    "Yaa Gyasi",
  ],
  "bookProvider": "Open Library",
  "categories": [
    "American literature",
    "nyt:combined-print-and-e-book-fiction=2020-09-20",
    "New York Times bestseller",
    "New York Times reviewed",
    "Brothers and sisters, fiction",
    "Fiction, family life",
    "Fiction, cultural heritage",
    "Fiction, family life, general",
    "Women medical students",
    "Fiction",
    "Ghanaians",
    "Grief",
    "Faith",
    "Mentally ill mothers",
    "Families",
    "Psychiatry",
    "Research",
    "Depressed persons",
    "Brothers",
    "Death",
    "Étudiantes en médecine",
    "Romans, nouvelles",
    "Chagrin",
    "Foi",
    "Mères vivant avec un trouble de santé mentale",
    "Familles",
    "Psychiatrie",
    "Recherche",
    "Frères",
    "Mort",
    "FICTION / Literary",
    "FICTION / Family Life / Siblings",
    "FICTION / Cultural Heritage",
  ],
  "description": "",
  "format": "book",
  "isbn": "9780385695176",
  "language": undefined,
  "link": "https://openlibrary.org/books/OL29999402M",
  "pageCount": undefined,
  "publishedDate": "Sep 02, 2020",
  "publisher": "Bond Street Books",
  "thumbnail": "https://covers.openlibrary.org/b/id/10415105-L.jpg",
  "title": "TRANSCENDENT KINGDOM",
}
`;
